### PR TITLE
WIP: script for olm e2e

### PR DESCRIPTION
**Pull request by pmorie**

### CODE REVIEW

File: scripts/olm-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+# otherwise:
+# - QUAY_ACCOUNT
+if [ -z "$KUBECONFIG" ]; then
+  echo "KUBECONFIG must be set"
+  exit 1
+fi
+
+# TODO: right way to do 'i-am-in-ci'
+IN_CI="${IN_CI:-""}"
+if [[ ! -z "$IN_CI" ]]; then
```
Comment (chancez) [inline scripts/olm-e2e.sh]: `if [[ -n "${IN_CI}" ]]`

File: scripts/olm-e2e.sh [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  echo "IMAGE_FORMAT=$IMAGE_FORMAT"
+  component="federation-controller"
+  pipeline_federation_image=${IMAGE_FORMAT/\$\{component\}/$component}
+  operator_registry_imagestream_tag="operator-registry:latest"
+  sed "s,quay.io/openshift/origin-federation-controller,$pipeline_federation_image," -i manifests/federation/0.0.5/federation.v0.0.5.clusterserviceversion.yaml
+  sed "s,quay.io/openshift/origin-federation-controller,$pipeline_federation_image," -i manifests/cluster-federation/0.0.5/cluster-federation.v0.0.5.clusterserviceversion.yaml
+
+  # build and push operator registry image, with the OLM CSV munged to reference
+  # the pipeline image for federation
+  oc new-build --name=operator-registry --dockerfile="$(<olm-testing/Dockerfile)"
```
Comment (chancez) [inline scripts/olm-e2e.sh]: This..is interesting.

**Comment by baijum** [comment]
For OLM integration testing, why did you go for CatalogSource with sourceType as "grpc" ( which requires to push the image to a registry) instead of using sourceType as "internal" (ConfigMap based)?
Example catalog source with configMap:
https://github.com/jarifibrahim/operator-dependency-poc/tree/master/catalogSource

**Comment by pmorie** [comment]
> For OLM integration testing, why did you go for CatalogSource with sourceType as "grpc" ( which requires to push the image to a registry) instead of using sourceType as "internal" (ConfigMap based)?

That was a mistake :) and I am going to be adopting configmap-based catalog sources. I found that once I tried to start doing e2e in CI that having to build the operator registry image just added way too many failure nodes. Configmap will be better :+1:

File: scripts/olm-e2e.sh [hunk 3]
```diff
@@ -0,0 +1,146 @@
+#!/bin/bash -e
+
+# olm-e2e.sh orchestrates deployent of the OLM manifests and images from this
```
Comment (SamiSousa) [inline scripts/olm-e2e.sh]: It's worth pointing out that the community-operators CI does all this (albeit with the end goal of running operator-sdk scorecard). Please take a look for inspiration (main script): https://github.com/operator-framework/community-operators/blob/master/scripts/ci/test-operator

**Pull request closed without merging**